docs/Conformance-Guide-draft.md

```diff
@@ -0,0 +1,103 @@
+- Required vs Best Practice: we currently distinguish if a criteria is a must-do for conformance versus if it is just a recommended best practice.
+- Goal: each criteria should in its description clearly state the goal or rationale for requiring it.
+- Testability: Each criteria should specify the way and extender as well as the Zowe Explorer squad can verify conformance.
+
+## General conformance criteria (0)
+
+These criteria are independent of the kind of extension produce and need to be fulfill for all cases.
+
+1. (Required) Naming: **If the extension uses the word "Zowe" in its name, it abides by Linux Foundation's Trademark Usage rules to ensure the word Zowe is used in a way intended by the Zowe community.**
+
+   The extender needs to provide a link where the offering will be announced and/or made available to the public.
+
+1. (Required): No Zowe CLI plugin installation requirement: **If the extender makes use of a Zowe CLI profile other than the Zowe Explorer default `zosmf` then the extension must not make any assumptions that a matching Zowe CLI plugin has been installed in the Zowe Explorer user's environment.**
+
+   In other words the extension must be full self-contained including all the code of the Zowe CLI Plugin that implements the new profile. This will not only simplify the end user experience, but also ensures that the extension can run in other VS Code compatible environments such as Cloud IDEs such as Eclipse Che. For VS Code extensions Zowe Explorer provides APIs to call that ensure that the users can store and access such new profiles in Zowe home directory folders and secure credentials store, which should be used or an equivalent needs to be provided. To test this requirement a user shall be able start the extension and use it without having Nodejs and Zowe CLI installed locally.
+
+1. (Required) Publication tag: **If the extension is published in a public catalog or marketplace such as Npmjs, Open-VSX, or VS Code Marketplace then it must use the tag or keyword "Zowe" so it can be found when searching for Zowe and be listed with other Zowe offerings.**
+
+   For example, if the extension is a VS Code extension then the "keywords" array of the package.json must include the entry "Zowe". This can be verified by looking at the page's Tags section in the VS Code Marketplace or if not yet present by extracting the vsix file and inspecting the package.json file.
+
+1. (Required) Support: **The extension needs to document how a user would get support or file issues for the extension for problems that are not related to Zowe and Zowe Explorer.**
+
+   The extender needs to document where the user can find this information in the conformance application.
+
+1. (Best Practice) User settings consistency: **For a consistent user experience we recommend that user settings and configuration settings follow the naming conventions as documented in the Zowe Explorer extensibility documentation.**
```

Comment:
Extension follows the recommended naming convention for the user settings and configuration settings indicated in the Zowe Explorer extensibility documentation

Reply:
If we take @jellypuno's suggested revision, I would propose adding the reason for this criteria ("to provide a consistent user experience") into the paragraph below.

Reply:
How do I add it and still make the sentence a Y or N question? Like this?

Extension provides a consistent user experience and follows the recommended naming convention for the user settings and configuration settings indicated in the Zowe Explorer extensibility documentation.